### PR TITLE
Change header map type to union type and add util functions and refactor test cases

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
$subject

### CODE REVIEW

File: grpc-ballerina/tests/10_grpc_ssl_client.bal [hunk 1]
```diff
@@ -53,11 +51,33 @@ public client class grpcMutualSslServiceClient {
...
+            headers = req.headers;
+        } else {
+            message = req;
+        }
+        var payload = check self.grpcClient->executeSimpleRPC("grpcservices.grpcMutualSslService/hello", message, headers);
+        [anydata, map<string|string[]>][result, respHeaders] = payload;
+        return {content: result.toString(), headers: respHeaders};
+    }
+
+}
```
Comment (BuddhiWathsala) [inline grpc-ballerina/tests/10_grpc_ssl_client.bal:83]:
```suggestion
}

```

File: grpc-ballerina/tests/11_grpc_byte_service.bal [hunk 2]
```diff
@@ -24,35 +24,43 @@ listener Listener ep11 = new (9101, {
...
         self.caller = caller;
     }
 
     public isolated function getId() returns int {
         return self.caller.getId();
     }
-
-    isolated remote function send(byte[] response) returns Error? {
+    
+    isolated remote function sendByte(byte[] response) returns Error? {
```
Comment (BuddhiWathsala) [inline grpc-ballerina/tests/11_grpc_byte_service.bal:43]: Shouldn't this be `sendBytes`?

File: grpc-ballerina/tests/23_server_streaming_with_record_client.bal [hunk 3]
```diff
@@ -49,8 +49,9 @@ public client class helloWorldServerStreamingClient {
...
 
-    isolated remote function lotsOfReplies(HelloRequest req, map<string[]> headers = {}) returns stream<anydata>|Error {
-        return self.grpcClient->executeServerStreaming("helloWorldServerStreaming/lotsOfReplies", req, headers);
+    isolated remote function lotsOfReplies(HelloRequest req) returns stream<anydata>|Error {
+        
+        return self.grpcClient->executeServerStreaming("helloWorldServerStreaming/lotsOfReplies", req);
     }
 
-}
+}
```
Comment (BuddhiWathsala) [inline grpc-ballerina/tests/23_server_streaming_with_record_client.bal:57]:
```suggestion
}

```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: grpc-ballerina/tests/11_grpc_byte_service.bal (hunk 2 above)
Comment (daneshk) [reply grpc-ballerina/tests/11_grpc_byte_service.bal]: Fixed in 3a0fabf

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**